### PR TITLE
add samples for enumerate & copy

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/samples/copy_blob.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        props = copied_blob.get_blob_properties()
+        status = props.copy.status
+        print("Copy status: " + status)
+        if status == "success":
+            # Copy finished
+            is_finished = True
+            break
+        time.sleep(10)
+
+    if not(is_finished):
```
Comment (bryevdv) [inline sdk/storage/azure-storage-blob/samples/copy_blob.py]: if you set `status = None` before the loop, this could check `if status != 'success':` and get rid of `is_finished` altogether. 

Either way, it seems like the operation could complete after 90 seconds is passed, and an abort will still be tried?

Comment (xiangyan99) [reply sdk/storage/azure-storage-blob/samples/copy_blob.py]: Yes. the copy operation may need more than 100s. I added this in order to demo how to abort the copy operation.

Comment (bryevdv) [reply sdk/storage/azure-storage-blob/samples/copy_blob.py]: @xiangyan99 My point is that the sleep happens at the *end* of a loop iteration, and so there is no check after 90s. If the job finished at 95s, it will be considered "not finished" even though it finished in under 100 seconds.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py [hunk 2]
```diff
@@ -84,8 +84,6 @@ class BlobClient(StorageAccountHostsMixin):  # pylint: disable=too-many-public-m
         account URL already has a SAS token. The value can be a SAS token string, an account
         shared access key, or an instance of a TokenCredentials class from azure.identity.
         If the URL already has a SAS token, specifying an explicit credential will take priority.
-    :keyword str secondary_hostname:
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py]: why is this removed?
